docs/learning/frontend/Extensions.md

```diff
@@ -0,0 +1,196 @@
+# iTwin.js Extensions
+
+An iTwin.js Extension is a separate JavaScript module that can load on demand into an iTwin.js frontend application.
+The separate deliverable enables Extensions to provide extensibility of an iTwin.js application at runtime.
+Extensions have access to a limited set of iTwin.js functionality through @itwin/core-extension to enable seamless integration with the host app.
+
+## What can extensions do?
+
+Extensions can be used for many different purposes, such as:
+- Add new a new [tool](./Tools.md) or [decorator](./ViewDecorations.md) to an existing application to better support your custom workflows.
+- Write event based processing, i.e., subscribe to an iModel Event, or Unified Selection Event, and process that change.
+
+## How to get started
+
+An iTwin.js Extension is just a javascript file and a manifest (i.e., a package.json file with some additional properties).
+To get started, create a new directory for the Extension.
+
+### Setup the Manifest
+The first step to creating an extension is to create a manifest.
+Create the package.json by running, in the directory created above:
+```
+npm init --yes
+```
+The following properties must be added to the package.json file:
+- _Name_: the name of the Extension.
+- _Version_: the version of the Extension in the format _x.x.x_.
+- _Main_: where to find the javascript file.
+- _ActivationEvents_: events that define when the iTwin.js application should load and run your Extension. Currently, we only support `onStartup`, which will execute the Extension as soon as it is loaded.
+
+Here is a minimal example:
+```json
+// package.json
+{
+  "name": "my-new-extension",
+  "version": "0.0.1",
+  "main": "./dist/index.js",
+  "type": "module",
+  "activationEvents": [
+    "onStartup"
+  ]
+}
+```
+
+Next, you'll want to add [TypeScript](https://www.typescriptlang.org/), and the required dependencies for developing with the iTwin.js shared libraries:
+```json
+  // package.json
+  "dependencies": {
+    "@itwin/core-extension": "^3.2.0"
+  },
+  "devDependencies": {
+    "typescript": "~4.4.0",
+    "@itwin/build-tools": "^3.2.0",
+  },
+```
+A basic tsconfig.json file needs to be setup for development. Create a new tsconfig.json file next to the package.json with the following contents:
+```json
+// tsconfig.json
+{
+  "extends": "./node_modules/@itwin/build-tools/tsconfig-base.json",
+  "include": ["./*.ts", "./*.tsx"]
+}
+```
+
+Next, add your favorite JavaScript tool to bundle your code together. Bundling is the process of combining multiple small source files into a single file. For JavaScript, popular bundlers are [rollup.js](https://rollupjs.org/guide/en/), [esbuild](https://esbuild.github.io/), and [webpack](https://webpack.js.org/). We recommend using esbuild:
```

Comment:
While reading, I was wondering what was mandatory and what was suggested...
I guess that bundling is required because only 1 .js file can be loaded, it might be good to mention that somewhere, regarding minify, iTwinJs is not minified itself, so I guess that is a suggestion ? (same thing for the polyfills, is that required by the `core-extensions` or is that just a convenience ?

Reply:
> While reading, I was wondering what was mandatory and what was suggested... I guess that bundling is required because only 1 .js file can be loaded, it might be good to mention that somewhere, regarding minify, iTwinJs is not minified itself, so I guess that is a suggestion ? (same thing for the polyfills, is that required by the `core-extensions` or is that just a convenience ?

Yeah bundling is required, I can mention that somewhere. Minify is a suggestion but a strong one, to keep extension size small. And I think you need the polyfills

Reply:
@raplemie let me know what you think of the changes

Reply:
Looks good for the why's.

To @aruniverse point, we can simply mention "Here is an example with esbuild:" if we dont feel confortable with "recommend", but that's just an idea :)